pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -0,0 +1,521 @@
+			}
+
+			sd := &tracingcfg.OpenCensusConfig{
+				StackdriverExporterEnabled: true,
+				StackdriverProjectId:       proj,
+				IncomingTraceContext:       allContexts,
+				OutgoingTraceContext:       allContexts,
+				StdoutExporterEnabled:      provider.Stackdriver.Debug,
+				TraceConfig: &opb.TraceConfig{
+					MaxNumberOfAnnotations:   200,
+					MaxNumberOfAttributes:    200,
+					MaxNumberOfMessageEvents: 200,
+				},
+			}
+
+			if meta.StsPort != "" {
+				stsPort, err := strconv.Atoi(meta.StsPort)
+				if err != nil || stsPort < 1 {
+					return nil, fmt.Errorf("could not configure Stackdriver tracer - bad sts port: %v", err)
+				}
+				sd.StackdriverGrpcService = &envoy_config_core_v3.GrpcService{
+					InitialMetadata: []*envoy_config_core_v3.HeaderValue{
+						{
+							Key:   "x-goog-user-project",
+							Value: proj,
+						},
+					},
+					TargetSpecifier: &envoy_config_core_v3.GrpcService_GoogleGrpc_{
+						GoogleGrpc: &envoy_config_core_v3.GrpcService_GoogleGrpc{
+							TargetUri:  "cloudtrace.googleapis.com",
+							StatPrefix: "oc_stackdriver_tracer",
+							ChannelCredentials: &envoy_config_core_v3.GrpcService_GoogleGrpc_ChannelCredentials{
+								CredentialSpecifier: &envoy_config_core_v3.GrpcService_GoogleGrpc_ChannelCredentials_SslCredentials{
+									SslCredentials: &envoy_config_core_v3.GrpcService_GoogleGrpc_SslCredentials{
+										RootCerts: &envoy_config_core_v3.DataSource{
+											Specifier: &envoy_config_core_v3.DataSource_Filename{
+												Filename: "/etc/ssl/certs/ca-certificates.crt",
+											},
+										},
+									},
+								},
+							},
+							CallCredentials: []*envoy_config_core_v3.GrpcService_GoogleGrpc_CallCredentials{
+								{
+									CredentialSpecifier: &envoy_config_core_v3.GrpcService_GoogleGrpc_CallCredentials_StsService_{
+										StsService: &envoy_config_core_v3.GrpcService_GoogleGrpc_CallCredentials_StsService{
+											TokenExchangeServiceUri: fmt.Sprintf("http://localhost:%d/token", stsPort),
+											SubjectTokenPath:        "/var/run/secrets/tokens/istio-token",
+											SubjectTokenType:        "urn:ietf:params:oauth:token-type:jwt",
+											Scope:                   "https://www.googleapis.com/auth/cloud-platform",
+										},
+									},
+								},
+							},
+						},
+					},
+				}
+			}
+
+			if provider.Stackdriver.MaxNumberOfAnnotations != nil {
+				sd.TraceConfig.MaxNumberOfAnnotations = provider.Stackdriver.MaxNumberOfAnnotations.Value
+			}
+			if provider.Stackdriver.MaxNumberOfAttributes != nil {
+				sd.TraceConfig.MaxNumberOfAttributes = provider.Stackdriver.MaxNumberOfAttributes.Value
+			}
+			if provider.Stackdriver.MaxNumberOfMessageEvents != nil {
+				sd.TraceConfig.MaxNumberOfMessageEvents = provider.Stackdriver.MaxNumberOfMessageEvents.Value
+			}
+			return anypb.New(sd)
+		})
+	default:
+		log.Warnf("unrecognized provider type: %T; will not configure tracing", provider)
```

Comment:
this seems a bit annoying as it will print the warning when using other extension providers (e.g. ext-authz).

Reply:
thanks for catching this. i removed it and turned down the logging levels throughout to reduce noise.